tests/unit/test_charm.py

```diff
@@ -121,4 +90,5 @@ def test_update_status_success(self, *_) -> None:
         self.harness.charm._stored.slurmctld_started = True
 
         self.harness.charm.on.update_status.emit()
-        self.assertEqual(self.harness.charm.unit.status, ActiveStatus("slurmd available"))
+        # Status should not change at all as slurmd should be active.
+        self.assertEqual(self.harness.charm.unit.status, MaintenanceStatus())
```

Comment:
I think ops.Harness just initializes the mocked charm with the default status of `MaintenanceStatus()`. I updated `_check_status(...)` so that only `_on_slurmd_started(...)` could set the status to `ActiveStatus()`, but I still want test coverage of the `_check_status(...)` method. Perhaps I can reword the comment to "ensure status is not modified by _check_status".

Reply:
Updated in commit https://github.com/omnivector-solutions/slurmd-operator/pull/21/commits/a423145aeac7b2a283dac6a4e5a59594adf6d463

Reply:
So the problem with this is that *if* the ops.Harness framework chooses to change the default status of `MaintenanceStatus()` this check will break for no reason.

A better assertion here might be something like the following:
```python3
prev_status = self.harness.charm.unit.status
self.harness.charm.on.update_status.emit()
...
self.assertEqual(self.harness.charm.unit.status, prev_status)
```

The above would indicate that the status hasn't changed as a result of the call to `_check_status` without a dependency on what the status was prior to the invocation.

Reply:
Great idea! Let's do this instead

Reply:
Implemented in https://github.com/omnivector-solutions/slurmd-operator/pull/21/commits/7b591ade7a36aab4f295c91b45792ebc1f077fff